src/data-type.js

```diff
@@ -1305,10 +1305,30 @@ const TYPE = module.exports.TYPE = {
       if (value instanceof Date) {
         return value;
       }
-      value = Date.parse(value);
-      if (isNaN(value)) {
+      var dateValue = Date.parse(value);
+      if (!isNaN(dateValue)) {
+        return dateValue;
+      }
+      var timespan = (function(input) {
```

Comment:
Is there any specific reason why this is in a nested function? Nesting functions like this might have an adverse impact on performance. I think the logic in here could easily be part of the enclosing function.